tests/test_debugging.py

```diff
@@ -376,3 +379,152 @@ def circuit():
             return qml.expval(qml.PauliZ(0))
 
         qml.snapshots(circuit)()
+
+
+# pylint: disable=protected-access
+class TestPLDB:
+    """Test the interactive debugging integration"""
+
+    def test_pldb_init(self):
+        """Test that PLDB initializes correctly"""
+        debugger = PLDB()
+        assert debugger.prompt == "[pldb]: "
+        assert getattr(debugger, "_PLDB__active_dev") == []
+
+    def test_valid_context_outside_qnode(self):
+        """Test that valid_context raises an error when breakpoint
+        is called outside of a qnode execution."""
+
+        with pytest.raises(TypeError, match="Can't call breakpoint outside of a qnode execution"):
+            with qml.queuing.AnnotatedQueue() as _:
+                qml.X(0)
+                qml.breakpoint()
+                qml.Hadamard(0)
+
+        def my_qfunc():
+            qml.X(0)
+            qml.breakpoint()
+            qml.Hadamard(0)
+            return qml.expval(qml.Z(0))
+
+        with pytest.raises(TypeError, match="Can't call breakpoint outside of a qnode execution"):
+            _ = my_qfunc()
+
+    def test_valid_context_not_compatible_device(self):
+        """Test that valid_context raises an error when breakpoint
+        is called outside of a qnode execution."""
+        dev = qml.device("default.mixed", wires=2)
+
+        @qml.qnode(dev)
+        def my_circ():
+            qml.X(0)
+            qml.breakpoint()
+            qml.Hadamard(0)
+            return qml.expva(qml.Z(0))
+
+        with pytest.raises(TypeError, match="Device not supported with breakpoint"):
+            _ = my_circ()
+
+        PLDB.reset_active_dev()
+
+    def test_add_device(self):
+        """Test that we can add a device to the global active device list."""
+        assert getattr(PLDB, "_PLDB__active_dev") == []
+
+        dev1, dev2, dev3 = (
+            qml.device("default.qubit", wires=3),
+            qml.device("default.qubit"),
+            qml.device("lightning.qubit", wires=1),
+        )
+
+        PLDB.add_device(dev1)
+        assert getattr(PLDB, "_PLDB__active_dev") == [dev1]
+
+        PLDB.add_device(dev2)
+        PLDB.add_device(dev3)
+        debugger_active_devs = getattr(PLDB, "_PLDB__active_dev")
+
+        for active_dev, d in zip(debugger_active_devs, [dev1, dev2, dev3]):
+            assert active_dev is d
+
+        PLDB.reset_active_dev()  # clean up the debugger active devices
+
+    dev_names = (
```

Comment:
What prevents us from supporting other devices?

Reply:
As long as its a simulator device, we should be able to support it, but for this MVP we are aimining to gurantee support for these two devices.